api/v1alpha1/project_config_types.go

```diff
@@ -342,6 +344,146 @@ type AzureWebhookReceiverConfig struct {
 	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
 }
 
+// GenericWebhookReceiverConfig describes a generic webhook receiver that can be
+// configured to respond to any arbitrary POST by applying user-defined actions
+// on user-defined sets of resources selected by name, labels and/or values in pre-built indices.
+// Both types of selectors support using values extracted from the request by
+// means of expressions. Currently, refreshing resources is the only supported
+// action and Warehouse is the only supported kind. "Refreshing" means
+// immediately enqueuing the target resource for reconciliation by its
+// controller. The practical effect of refreshing a Warehouses is triggering its
+// artifact discovery process.
+type GenericWebhookReceiverConfig struct {
+	// SecretRef contains a reference to a Secret. For Project-scoped webhook
+	// receivers, the referenced Secret must be in the same namespace as the
+	// ProjectConfig.
+	//
+	// For cluster-scoped webhook receivers, the referenced Secret must be in the
+	// designated "cluster Secrets" namespace.
+	//
+	// The Secret's data map is expected to contain a `secret` key whose value
+	// does NOT need to be shared directly with the sender. It is used only by
+	// Kargo to create a complex, hard-to-guess URL, which implicitly serves as a
+	// shared secret.
+	//
+	// +kubebuilder:validation:Required
+	SecretRef corev1.LocalObjectReference `json:"secretRef" protobuf:"bytes,1,opt,name=secretRef"`
+
+	// Actions is a list of actions to be performed when a webhook event is received.
+	//
+	// +kubebuilder:validation:MinItems=1
+	Actions []GenericWebhookAction `json:"actions,omitempty" protobuf:"bytes,2,rep,name=actions"`
+}
+
+// GenericWebhookAction describes an action to be performed on a resource
+// and the conditions under which it should be performed.
+type GenericWebhookAction struct {
+	// Name is the name of the action to be performed. `Refresh` is the only
+	// action currently supported.
+	//
+	// +kubebuilder:validation:Enum=Refresh;
+	Name GenericWebhookActionName `json:"action" protobuf:"bytes,1,opt,name=action"`
```

Comment:
I don't think `Name` is the ideal name for this field. Everywhere else in Kargo (or Kubernetes, for that matter) that I can think of, a "name" is an identifier.

A better name for this field may be `Kind`. To go along with that, I'd also suggest renaming the enumerated `GenericWebhookActionName` type to `GenericWebhookActionKind`.